modules/ROOT/pages/distributed-locking.adoc

```diff
@@ -0,0 +1,104 @@
+= Distributed Locking
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: distributed locking, cluster
+
+Mule runtime engine provides the ability to create locks for synchronizing access to resources within Mule components. To manage concurrent access to resources, Mule provides a lock factory that you can access programmatically by scripts or in custom extensions built with the xref:1.1@mule-sdk::getting-started.adoc[Java SDK].
+
+Any locks you create with Mule's lock factory work seamlessly on deployment models which use a single server or a cluster of servers. This means that if you have a server running the same flow in multiple threads or a cluster environment running the same app, you can guarantee resource synchronization with Mule locks. Additionally, Mule's locking system offers a simple API to access shared locks. Learn how to create locks and sharing them between operations in the following sections.
+
+
+== Creating a Lock
+
+To create a lock, follow these tasks:
+
+. Gain access to a `LockFactory` through injection. +
+If you are building an extension with the Java SDK and need locking for one of your  operations, you need to inject the `LockFactory` into your class. +
+. Use the `LockFactory` to start creating locks. +
+Refer to the following sample code which declares an extension operation that manages code access through Mule locks and thus prevents synchronization issues when the operation is executing in parallel (like in a scatter gather scope or in different cluster nodes). +
+. Use an identifier for the lock you create. +
+The identifier allows you to access the same lock from different threads without having to explicitly share the same lock instance between them. +
+
+This lock implements the `java.util.concurrent.Lock` interface, so calling the method `lock` will make the thread wait until it acquires the lock. +
+
+After obtaining the lock, the critical code section can be executed and, no matter what the result is, we want to release the obtained lock (to prevent deadlocks on other threads waiting for this lock), so we wrap the `lock.unlock` call in a `finally` block.
+
+[source,java,linenums]
+----
+public class ClusterOperations {
+
+  @Inject
+  private LockFactory lockFactory;
+
+  public void sharedResourceOperation() {
+      Lock lock = lockFactory.createLock("sharedResourceId");
+      lock.lock();
+      try {
+          // Lock acquired, execute critical code.
+      } finally {
+          lock.unlock();
+      }
+  }
+}
+----
+
+If you are using the scripting extension, you must first access the registry to obtain the `LockFactory` and then create locks through scripting code. The following example uses Groovy to achieve the same result as the previous example: obtain the  `LockFactory` (this time through the `Registry`), create a new `Lock` for the given ID, wait until obtaining the lock to execute critical code and after that release the lock.
```

Comment:
Do we need to identify to what the scripting extension is an extension? "...using the scripting extension to <what>...".
```suggestion
If you are using the scripting extension, you must first access the registry to obtain the lock factory and then create locks through scripting code. The following example uses Groovy:
```